### PR TITLE
pix2pix: Always same test_dataset picture 

### PR DESCRIPTION
**Pull request by crypdra**
- The shuffle operation had been applied on the train_dataset for a second time.
- The .take(1) operation will always return the same element, as it does not iterate the dataset

I fixed the dataset shuffle operation and replaced the .take(1) problem by defining an iterator. In order for the iterator to work for more than 107 epochs I defined a repeat operation.

### CODE REVIEW

**Comment by crypdra** [comment]
@googlebot I signed it!

**Comment by yashk2810** [comment]
The image is kept same because it allows us to see the growth of the model. You can compare the picture from epoch 30 to epoch 15 and see the progress. If the image keeps changing, its hard to see the progress.

**Comment by crypdra** [comment]
> The image is kept same because it allows us to see the growth of the model. You can compare the picture from epoch 30 to epoch 15 and see the progress. If the image keeps changing, its hard to see the progress.

But there's a comment in the code which explicitly says, that the code is supposed to change the test picture on every epoch:
```
# shuffling so that for every epoch a different image is generated
# to predict and display the progress of our model.
```

**Comment by yashk2810** [comment]
> But there's a comment in the code which explicitly says, that the code is supposed to change the test image on every epoch:

Sorry about that. I fixed it here (https://github.com/tensorflow/docs/commit/3b597c2492e626cf5c79ba4a97f19e0fb1ff0383)

**Comment by lamberta** [comment]
Sounds like this is resolved. Thanks, @crypdra, for raising the issue, and thanks, @yashk2810, for the fix.
